### DISCUSSION TITLE
Shiny.BluetoothLE - Scan for Arduino/Rasberry PI BLE

### DISCUSSION THREAD

**Discussion by RobSchein** [shinyorg/shiny#1414] [Q&A]
Thanks so much for the library, very handy and I'm surprised (and pleased) to have found something I believe will short-circuit my develoment!  Using Maui...In following your Getting Started guide, when I Scan().Subscribe(...), what returns is a list of what looks like beacons (AdvertizementData.LocalName...I get a bunch of gDevice-beacon or beac names).  I have a little arduino device that is advertising and shows up in out-of-box android scanning tools, but I don't see it listed in the Shiny scanned list of devices.  In BLE land (especially Arduino), there is the concpt of Central/Peripheral devices.  My device is a Central, and your guide references Peripherals.  I'm wondering whether I am confused with the terminology.  In any event, I have that Central device and I want to interact with it.  Thank you in advance for your advice!

**Comment by aritchie** [comment]
This library will only scan for BLE based devices, beyond that there are no special filters

**Comment by RobSchein** [comment]
Thanks for your fast reply, Alan!  This is a BLE device (not BT classic), that’s why I was surprised not to see it show up in the list of scanned devices.  Not exactly sure where to take it from here.

 



Thanks,

-Rob

**Comment by RobSchein** [comment]
While on PTO, my idea bubble triggered.  Debugging directly on an android phone instead of a Visual Studio android emulator did the trick.  Since I'm just scratching the surface of mobile development, not sure why the emulator is not behaving like the hardware does.  I'll have to do more research.  At least I am making progress now.

**Reply by aritchie** [reply]
The android & ios emulators don't support bluetooth at all.
